src/PKSim.Core/Services/IndividualMoleculeFactory.cs

```diff
@@ -128,7 +131,10 @@ protected TMolecule CreateMolecule(string moleculeName)
          if (!string.IsNullOrEmpty(groupName))
             parameter.GroupName = groupName;
 
+         //Because we update the formula of the parameter, we need to make sure we also reset the formulaId so that it will appear as being unique
          parameter.Formula.ReplaceKeywordsInObjectPaths(new[] {ObjectPathKeywords.MOLECULE}, new[] {moleculeName});
+         parameter.Formula.Id = _idGenerator.NewId();
```

Comment:
@Yuri05 This was the error here. We have a formula that we get from the DB but we modify it "by hand". I need to update the Id otherwise it will be serialized as ONE and on load, you get the wrong references...